WordPress/Classes/ViewRelated/Reader/Sorting/ReaderSortingOptionSheetViewController.swift

```diff
@@ -0,0 +1,277 @@
+import UIKit
+
+struct ReaderSortingActionSheetOption {
+    let title: String
+    let image: UIImage
+    let checked: Bool
+    let identifier: String
+    let sortingOption: ReaderSortingOption
+
+    init(title: String, image: UIImage, checked: Bool, identifier: String, sortingOption: ReaderSortingOption) {
+        self.title = title
+        self.image = image
+        self.checked = checked
+        self.identifier = identifier
+        self.sortingOption = sortingOption
+    }
+}
+
+class ReaderSortingActionSheetOptionControl: ClosureControl {
+    enum Constants {
+        static let iconsHeight: CGFloat = 24.0
+        static let iconsWidth: CGFloat = 24.0
+        static let iconLeading: CGFloat = 16.0
+        static let labelLeading: CGFloat = 6.0
+        static let minimalTappableHeight: CGFloat = 44.0
+        // TODO: check if we should color from assets or from hex
+        static let checkTintColor: UIColor = UIColor(light: UIColor(named: "Blue50") ?? UIColor(hexString: "2271B1"), dark: UIColor(named: "Blue30") ?? UIColor(hexString: "5198D9"))
+        // TODO: check if we can use iOS system colors here
+        static let iconTintColor: UIColor = UIColor(light: UIColor(hexString: "4D4D4D"), dark: UIColor(hexString: "BFBFBF"))
+        static let labelColor: UIColor = UIColor(light: .black, dark: .white)
+    }
+
+    private lazy var checkView: UIImageView = {
+        let view = UIImageView()
+        view.image = .gridicon(.checkmark)
+        view.tintColor = Constants.checkTintColor
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private lazy var imageView: UIImageView = {
+        let view = UIImageView()
+        view.tintColor = Constants.iconTintColor
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private lazy var label: UILabel = {
+        let view = UILabel()
+        view.textColor = Constants.labelColor
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private(set) var option: ReaderSortingOption = .noSorting
+
+    override var isSelected: Bool {
+        didSet {
+            checkView.isHidden = !isSelected
+        }
+    }
+
+    // MARK: - setup
+
+    convenience init(option: ReaderSortingActionSheetOption, closure: @escaping () -> Void) {
+        self.init(frame: CGRect.zero, minimalTappableHeight: Constants.minimalTappableHeight, closure: closure)
+
+        self.option = option.sortingOption
+        isSelected = option.checked
+        imageView.image = option.image
+        label.text = option.title
+        accessibilityIdentifier = option.identifier
```

Comment:
I plan to go thru all UI components in final PR to use proper accesibility identifiers and labels.